staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -54,35 +113,49 @@ func NewNamedDelayingQueue(name string) DelayingInterface {
 // NewDelayingQueueWithCustomClock constructs a new named workqueue
 // with ability to inject real or fake clock for testing purposes
 func NewDelayingQueueWithCustomClock(clock clock.WithTicker, name string) DelayingInterface {
-	return newDelayingQueue(clock, NewNamed(name), name)
+	return newDelayingQueue(clock, NewNamed(name), name, nil)
 }
 
-func newDelayingQueue(clock clock.WithTicker, q Interface, name string) *delayingType {
+func newDelayingQueue(clock clock.WithTicker, q Interface, name string, log queueLogger) *delayingType {
 	ret := &delayingType{
-		Interface:       q,
-		clock:           clock,
-		heartbeat:       clock.NewTicker(maxWait),
-		stopCh:          make(chan struct{}),
-		waitingForAddCh: make(chan *waitFor, 1000),
-		metrics:         newRetryMetrics(name),
+		Interface:          q,
+		clock:              clock,
+		nextReadyAtTimerCh: make(<-chan time.Time),
+		heartbeat:          clock.NewTicker(maxWait),
+		stopCh:             make(chan struct{}),
+		waitingForAddCh:    make(chan *waitFor, 1000),
+		metrics:            newRetryMetrics(name),
+		logger:             log,
+		waitForQueueFilled: make([]chan int, 0),
```

Comment:
assigning nil would be more appropriate here

Reply:
ok changed